esmvaltool/cmor/fix.py

```diff
@@ -80,18 +82,37 @@ def fix_metadata(cube, short_name, project, dataset, cmor_table=None,
         If the checker detects errors in the metadata that it can not fix.
 
     """
-    for fix in Fix.get_fixes(
-            project=project, dataset=dataset, variable=short_name):
-        cube = fix.fix_metadata(cube)
-    if cmor_table and mip:
-        checker = _get_cmor_checker(
-            table=cmor_table,
-            mip=mip,
-            short_name=short_name,
-            fail_on_error=False,
-            automatic_fixes=True)
-        checker(cube).check_metadata()
-    return cube
+    fixes = Fix.get_fixes(
+        project=project, dataset=dataset, variable=short_name
+    )
+    fixed_cubes = []
+    by_file = defaultdict(list)
+    for cube in cubes:
+        by_file[cube.attributes['source_file']].append(cube)
```

Comment:
I would change this to
```suggestion
        by_file[cube.attributes.get('source_file', '')].append(cube)
```
so the function can also be used with cubes that do not have the 'source_file' attribute